packages/plugin-earlybirds/src/actions/earlybirdAction.ts

```diff
@@ -0,0 +1,140 @@
+import { ModelProviderName } from "@moxie-protocol/core";
+import { DuneClient, QueryParameter, RunQueryArgs } from "@duneanalytics/client-sdk";
+import { generateUniqueGroupName } from "@moxie-protocol/plugin-moxie-groups/src/utils";
+
+const client = new DuneClient(process.env.DUNE_API_KEY!);
+
+export const earlybirdAction: Action = {
+    name: "EARLYBIRD",
+    similes: ["EARLY_BUYERS", "EARLY_BIRDS"],
+    description: "Get the earlybirds of a set of tokens",
+    suppressInitialMessage: true,
+    validate: async () => true,
+    handler: async (runtime: IAgentRuntime, message: Memory, state: State, _options: { [key: string]: unknown }, callback: HandlerCallback) => {
+        const traceId = message.id;
+        const moxieUserInfo = state.moxieUserInfo as MoxieUser;
+        const moxieUserId = moxieUserInfo.id;
+
+        try {
+            elizaLogger.debug(traceId, `[EarlybirdAction] [${moxieUserId}] Starting Earlybird calculation`);
+            const latestMessage = message.content.text;
+            const context = composeContext({
+                state: {
+                    ...state,
+                    latestMessage: latestMessage,
+                    moxieUserId,
+                },
+                template: earlyBirdsTemplate,
+            });
+
+            const earlybirdsResponse = await await generateObjectDeprecated({
+                runtime,
+                context: context,
+                modelClass: ModelClass.MEDIUM,
+                modelConfigOptions: {
+                    modelProvider: ModelProviderName.OPENAI,
+                    temperature: 0.0,
+                    apiKey: process.env.OPENAI_API_KEY!,
+                    modelClass: ModelClass.MEDIUM
+                }
+            });
+
+            elizaLogger.debug(traceId, `[EarlybirdAction] earlybirdsResponse: ${JSON.stringify(earlybirdsResponse)}`);
+            const queryId = 5802474;
+            const opts: RunQueryArgs = {
+                queryId,
+                query_parameters: [
+                    QueryParameter.number("n_buyers", 200),
+                    QueryParameter.text("token1", earlybirdsResponse.token1.address),
+                    QueryParameter.text("token2", earlybirdsResponse.token2.address),
+                    QueryParameter.text("token3", earlybirdsResponse.token3.address),
+                    QueryParameter.text("token4", earlybirdsResponse.token4.address),
+                ]
+            }
+            const result = await client.runQuery(opts);
+            elizaLogger.debug(traceId, `[EarlybirdAction] result: ${JSON.stringify(result)}`);
+            const earlybirds = result.result.rows;
+            if (earlybirds.length === 0) {
+                await callback({
+                    text: `No earlybirds found for the tokens ${earlybirdsResponse.token1.address}, ${earlybirdsResponse.token2.address}, ${earlybirdsResponse.token3.address}, ${earlybirdsResponse.token4.address}`,
```

Comment:
Let's also make the length here dynamic depending on the AI response + use the `$[tokenSymbol|tokenAddress]` format